src/lints/clone_on_copy.rs

```diff
@@ -0,0 +1,71 @@
+use crate::context::{CairoLintKind, Lint};
+use crate::queries::{get_all_function_bodies, get_all_function_calls};
+use cairo_lang_defs::ids::{LanguageElementId, ModuleItemId};
+use cairo_lang_defs::plugin::PluginDiagnostic;
+use cairo_lang_diagnostics::Severity;
+use cairo_lang_semantic::db::SemanticGroup;
+use cairo_lang_semantic::ExprFunctionCall;
+use cairo_lang_syntax::node::TypedStablePtr;
+use itertools::Itertools;
+
+const T_COPY_CLONE: &str = "core::clone::TCopyClone";
+
+pub struct CloneOnCopy;
+
+/// ## What it does
+///
+/// Checks for usage of .clone() on a Copy type.
+///
+/// ## Example
+///
+/// ```cairo
+///     let a: felt252 = 'Hello';
+///     let b = a.clone()
+/// ```
+impl Lint for CloneOnCopy {
+    fn allowed_name(&self) -> &'static str {
+        "clone_on_copy"
+    }
+
+    fn diagnostic_message(&self) -> &'static str {
+        "using `clone` on type which implements Copy trait"
+    }
+
+    fn kind(&self) -> CairoLintKind {
+        CairoLintKind::CloneOnCopy
+    }
+}
+
+pub fn check_clone_on_copy(
+    db: &dyn SemanticGroup,
+    item: &ModuleItemId,
+    diagnostics: &mut Vec<PluginDiagnostic>,
+) {
+    let Ok(_) = item.module_file_id(db.upcast()).file_id(db.upcast()) else {
+        return;
+    };
+
+    let function_bodies = get_all_function_bodies(db, item);
+    for function_body in function_bodies.iter() {
+        let function_call_exprs = get_all_function_calls(function_body);
+        for function_call_expr in function_call_exprs {
+            check_clone_usage(db, &function_call_expr, diagnostics);
+        }
+    }
+}
+
+fn check_clone_usage(
+    db: &dyn SemanticGroup,
+    expr: &ExprFunctionCall,
+    diagnostics: &mut Vec<PluginDiagnostic>,
+) {
+    let function_name = expr.function.full_path(db).split("::").take(3).join("::");
```

Comment:
Do we really need to split it? Shouldn't the path always be just `core::clone::TCopyClone`?